new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiConfigMapOverride.java

```diff
@@ -0,0 +1,679 @@
+    checkServerReadyStatusByExec(adminServerPodName, domainNamespace);
+
+    // check managed server pods re-started sequentially
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Wait for managed server pod {0} to be restarted in ns {1}",
+          managedServerPrefix + i, domainNamespace);
+      checkPodCreated(managedServerPrefix + i, domainUid, domainNamespace);
+      checkPodReady(managedServerPrefix + i, domainUid, domainNamespace);
+      checkServiceCreated(managedServerPrefix + i, domainNamespace);
+    }
+
+    String newAdminPodCreationTime = getPodCreationTimestamp(domainNamespace, "", adminServerPodName);
+    assertNotNull(newAdminPodCreationTime, "adminPodCreationTime returns NULL");
+    logger.info("NewAdminPodCreationTime {0} ", newAdminPodCreationTime);
+    if (Long.parseLong(newAdminPodCreationTime) == Long.parseLong(adminPodCreationTime)) {
+      logger.info("NewAdminPodCreationTime {0} BeforeAdminPodCreationTime ${1}",
+          newAdminPodCreationTime, adminPodCreationTime);
+      fail("New pod creation time must be later than the original timestamp");
+    }
+    oracle.weblogic.kubernetes.utils.ExecResult result = null;
+    int adminServiceNodePort = getAdminServiceNodePort(adminServerPodName + "-external", null, domainNamespace);
+    try {
+      checkJdbc = new StringBuffer("status=$(curl --user weblogic:welcome1 ");
+      checkJdbc.append("http://" + K8S_NODEPORT_HOST + ":" + adminServiceNodePort)
+          .append("/management/wls/latest/datasources/id/TestDataSource/")
+          .append(" --silent --show-error ")
+          .append(" -o /dev/null")
+          .append(" -w %{http_code});")
+          .append("echo ${status}");
+      logger.info("CURL command {0}", new String(checkJdbc));
+      result = exec(new String(checkJdbc), true);
+    } catch (Exception ex) {
+      logger.info("Caught unexpected exception {0}", ex);
+      fail("Got unexpected exception" + ex);
+    }
+
+    logger.info("Curl command returns {0}", result.toString());
+    assertEquals("200", result.stdout(), "Datasource configuration not found");
+    logger.info("Found the DataSource configuration ");
+
+  }
+
+  // This method is needed in this test class, since the cleanup util
+  // won't cleanup the images.
+  @AfterAll
+  void tearDown() {
+
+    // Delete domain custom resource
+    logger.info("Delete domain custom resource in namespace {0}", domainNamespace);
+    assertDoesNotThrow(() -> deleteDomainCustomResource(domainUid, domainNamespace),
+        "deleteDomainCustomResource failed with ApiException");
+    logger.info("Deleted Domain Custom Resource " + domainUid + " from " + domainNamespace);
+
+    // delete the domain image created for the test
+    if (miiImage != null) {
+      deleteImage(miiImage);
+    }
+
+  }
+
+  private String createImageAndVerify() {
+    // create unique image name with date
+    DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
```

Comment:
i see this enough, that maybe it could be refactored

Reply:
Boilerplate code from MiiDomain.java. May we should move to a common utility class.